resources/todomvc/vanilla-examples/javascript-web-components/src/components/todo-app/todo-app.component.js

```diff
@@ -0,0 +1,147 @@
+import template from "./todo-app.template.js";
+import { useRouter } from "../../hooks/useRouter.js";
+
+import globalStyles from "../../styles/global.constructable.js";
+import appStyles from "../../styles/app.constructable.js";
+import mainStyles from "../../styles/main.constructable.js";
+class TodoApp extends HTMLElement {
+    #isReady = false;
+    #data = [];
+    constructor() {
+        super();
+
+        const node = document.importNode(template.content, true);
+        this.topbar = node.querySelector("todo-topbar");
+        this.list = node.querySelector("todo-list");
+        this.bottombar = node.querySelector("todo-bottombar");
+
+        this.shadow = this.attachShadow({ mode: "open" });
+        this.htmlDirection = document.querySelector("html").getAttribute("dir") || "ltr";
+        this.shadow.host.setAttribute("dir", this.htmlDirection);
```

Comment:
I believe `this.shadow.host` is exactly the same as `this`.
```suggestion
        this.setAttribute("dir", this.htmlDirection);
```